### PR TITLE
refactored XlatStr to avoid buffer overruns

### PR DESCRIPTION
**Pull request by vitaliy-aksyonov**

### CODE REVIEW

File: goldlib/gall/gespell.cpp [hunk 1]
```diff
@@ -423,8 +424,7 @@ void CMSSpellLang::BuildSuggest(const char *text, CSpellSuggestV &suggest)
         if (flag && mSZ[idx])
         {
             flag = false;
-            RecodeText(&mSZ[idx], &mSZ[idx], false);
-            suggest.push_back(std::pair<byte, std::string>(0, "  " + std::string(&mSZ[idx]) + char(' ')));
+            suggest.push_back("  " + RecodeText(&mSZ[idx], false)) + char(' '));
```
Comment (shtirlic) [inline goldlib/gall/gespell.cpp:427]:
```suggestion
            suggest.push_back("  " + RecodeText(&mSZ[idx], false) + char(' '));
```

File: golded3/geedit.cpp [hunk 2]
```diff
@@ -2330,7 +2330,7 @@ void IEclass::SCheckerMenu()
         finaltag = menu.Run(schecker, buff);
         if (finaltag >= 0)
         {
-            std::string &str = schecker.GetSuggest()[finaltag].second;
```
Comment (spanevin) [inline golded3/geedit.cpp:2333]: Hmm, why it was ".second" before and now it's removed?

Comment (vitaliy-aksyonov) [reply golded3/geedit.cpp:2333]: It was vector of pairs. And first in the pair never used.

Comment (spanevin) [reply golded3/geedit.cpp:2333]: Why it was never used? :) Maybe there were reasons?

Comment (vitaliy-aksyonov) [reply golded3/geedit.cpp:2333]: I mean that member not used anywhere in the code. Both for MS and hunspell.

Comment (spanevin) [reply golded3/geedit.cpp:2333]: It is a tuple of <byte, string>. If I understand it correctly, &str was assigned to the string part of the tuple before, now it points the the beginning of the tuple, isn't it? Shouldn't we compensate the extra byte (the first item of the tuple) in the line 2335 or somehow else? Sorry I I'm writing something stupid, c++ is not my strong part...

Comment (vitaliy-aksyonov) [reply golded3/geedit.cpp:2333]: Just looked at commit, when spell checker was added. Maybe author was going to use it for ordering, but all inserts into vector has that byte set to 0 and no reads for that field. There is no reason to keep that field.

Comment (spanevin) [reply golded3/geedit.cpp:2333]: so, in order to keep functionality as it was before and don't break it with some unexpected 0 at the beginning of some buffer, shouldn't we replace the tuple with just string?

Comment (vitaliy-aksyonov) [reply golded3/geedit.cpp:2333]: It's not a pure c. :) That 0 byte was never used.
In std::pair<byte, string> both fields are totally separate members.

Comment (spanevin) [reply golded3/geedit.cpp:2333]: Sorry, I haven't found that you actually replaced the type implementation behind. I looked again, and now I see it. No questions anymore :)

**Review by spanevin — APPROVED** [review]

**Pull request merged**